router/router.go

```diff
@@ -785,10 +785,10 @@ func (worker *workerT) processDestinationJobs() {
 		_destinationJob := destinationJob
 
 		for _, destinationJobMetadata := range _destinationJob.JobMetadataArray {
-			handledJobMetadatas[destinationJobMetadata.JobID] = &destinationJobMetadata
+			_destinationJobMetadata := destinationJobMetadata
```

Comment:
[optional] It should fine, right? This way it is more idiomatic.

```suggestion
			destinationJobMetadata := destinationJobMetadata
```

Alternatively we could even avoid creating a copy and do:

```go
for i := range _destinationJob.JobMetadataArray {
    handledJobMetadatas[destinationJobMetadata.JobID] = &_destinationJob.JobMetadataArray[i]
    // ...
```